marker_lints/src/not_using_has_span_trait.rs

```diff
@@ -0,0 +1,44 @@
+use marker_api::prelude::*;
+
+marker_api::declare_lint! {
+    /// ### What it does
+    /// Suggests using `impl HasSpan` for functions that need to take `Span` as
+    /// a parameter to make them more ergonomic.
+    NOT_USING_HAS_SPAN_TRAIT,
+    Warn,
+}
+
+pub(crate) fn check_item<'ast>(cx: &'ast MarkerContext<'ast>, item: ItemKind<'ast>) {
+    let ItemKind::Fn(func) = item else { return };
+
+    for param in func.params() {
+        let ast::TyKind::Path(path) = strip_references(param.ty()) else {
+            continue;
+        };
+
+        let ident = path
+            .path()
+            .segments()
+            .last()
+            .unwrap_or_else(|| panic!("BUG: path must have at least one segment, but got: {path:#?}"))
+            .ident();
+
+        if ident != "Span" {
+            continue;
+        }
+
+        cx.emit_lint(
+            NOT_USING_HAS_SPAN_TRAIT,
+            func,
+            "use impl HasSpan instead of Span for more flexibility",
+        )
+        .span(param.ty().span());
```

Comment:
Same problem as in https://github.com/rust-marker/marker/issues/275. I couldn't use the `param.ty()` as the emission node directly, unfortunately.

Reply:
Yes, I plan to make the ast types emission nodes as well :)